src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,75 @@
-export const Pagination = () => {};
+import React from 'react';
+
+type PaginationProps = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const pages = [];
+
+  for (let i = 1; i <= totalPages; i++) {
+    pages.push(i);
+  }
+
+  const handlePageChange = (page: number) => {
+    if (page !== currentPage && page > 0 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
```

Comment:
The 'handlePageChange' function is well-implemented with checks to prevent unnecessary calls to 'onPageChange'.